tests/multinode.rs

```diff
@@ -631,94 +631,102 @@ fn test_multi_node_dynamic_network() {
 
     let mut validators: Vec<_> = t2.into_iter().map(|t| t.join().unwrap()).collect();
 
-    let now = Instant::now();
+    let mut client = mk_client(&leader_data);
+    let mut last_finality = client.get_finality();
+    info!("Last finality {}", last_finality);
+    let start = Instant::now();
     let mut consecutive_success = 0;
-    let mut failures = 0;
     let mut max_distance_increase = 0i64;
+    let mut expected_balance = leader_balance;
     for i in 0..std::cmp::max(20, num_nodes) {
-        //verify leader can do transfer
-        let expected = ((i + 3) * 500) as i64;
-        let leader_balance = retry_send_tx_and_retry_get_balance(
-            &leader_data,
-            &alice_arc.read().unwrap(),
-            &bob_pubkey,
-            Some(expected),
-        ).unwrap();
-        if leader_balance != expected {
-            info!(
-                "leader dropped transaction {} {:?} {:?}",
-                i, leader_balance, expected
+        trace!("getting leader last_id");
+        let last_id = client.get_last_id();
+        trace!("executing leader transfer");
+        let sig = client
+            .transfer(
+                500,
+                &alice_arc.read().unwrap().keypair(),
+                bob_pubkey,
+                &last_id,
+            )
+            .unwrap();
+
+        expected_balance += 500;
+
+        assert!(client.poll_for_signature(&sig).is_ok());
+
+        let now = Instant::now();
+        let mut finality = client.get_finality();
+        while finality == last_finality {
```

Comment:
```
while duration_as_ms(now.elapsed()) < client.get_finality() {
               sleep(Duration::from_millis(100));
}
```
right?  if finality is longer then T since confirmation we wait.  If its shorter that means 2/3rds of the validators have moved passed T.